[python] Fix toggle breakpoint behavior when invoked twice in a row

sunlin7 opened the pull request:
Currently pressing `, d b` can toggle debug line on a *.py file, but the cursor will be moved, so user have to move cursor back to the target line then to do "Toggle" again.

This PR will keep cursor on the same line, so user can press `, d b` to do a really "Toggle" operation (eg: press `, d b` twice).

bcc32 commented on layers/+lang/python/funcs.el:
```diff
@@ -205,11 +205,11 @@ ROOT-DIR should be the path for the environemnt, `nil' for clean up"
+      (insert trace ?\n)
+      (python-indent-line)
+      (forward-line -1))
+    (mwim-beginning-of-code-or-line)))
```
This command is part of the `mwim` package, not enabled by default.

bcc32 reviewed (changes requested):
It seems weird to leave point on the debugging line rather than the line where the user originally invoked `,db`.  Perhaps we should just make `,db` also attempt to search the line before the cursor for the previously-inserted debug statement, and delete it if found?

sunlin7 commented:
@bcc32  I agree the behavior on this patch is not perfect.
But if we search backward and remove previous line, it still not like a "Toggle" operation. 
Only keep the cursor on same line can act like a "Toggle".
And I removed the `mwim-beginning-of-code-or-line`.

bcc32 committed:
spacemacs/python-toggle-breakpoint: Also check previous line

This allows the command, when inserting a breakpoint, to keep the
cursor on the original line where it was invoked, but still be able to
toggle "off" the breakpoint on either the original line or the
breakpoint line.

bcc32 commented:
I was thinking we could have the command check the current line *and* the previous line.  Pushed a commit to show what I mean, let me know if you think that makes sense.

bcc32 committed:
spacemacs/python-toggle-breakpoint: Use string-search for literal strings

Avoid interpreting `trace` as a regexp.

sunlin7 commented:
@bcc32 It's good for me with the behavior to try check previous line. Thanks

bcc32 commented:
Thanks for the PR!

sunlin7 committed:
* layers/+lang/python/funcs.el: Keep cursor on same line when toggle debug line
